### PR TITLE
fix: update projectUrl and add more logs

### PR DESCRIPTION
**Pull request by adong**
## Context

1) Use `{ projectUrl }` instead
2) Only update `projectUrl` if not using self host sonar
3) Add more logs to track which lines of code are failing

## License
I confirm that this contribution is made under a BSD license and that I have the authority necessary to make this contribution on behalf of its copyright owner.

### CODE REVIEW

File: plugins/coverage/token.js [hunk 1]
```diff
@@ -82,50 +84,57 @@ module.exports = config => ({
...
+            if (!pipeline && pipelineId) {
+                pipeline = await pipelineFactory.get(pipelineId);
+
+                logger.info(`looking up again, pipeline:${pipelineId}, and found pipeline: ${pipeline}`);
+            }
+
+            if (pipeline && projectUrl) {
+                try {
+                    const pipelineSonarBadge = {
+                        defaultName: `${pipelineId}`, // ensure pipelineId is stored as String instead of Integer
```
Comment (adong) [inline plugins/coverage/token.js:103]: https://github.com/screwdriver-cd/data-schema/blob/2c89a71805b5700cff322a87984e2e94f5f0b61e/models/pipeline.js

**Review by klu909 — APPROVED** [review]

File: plugins/coverage/token.js [hunk 2]
```diff
@@ -82,50 +84,57 @@ module.exports = config => ({
...
+                        defaultUri: projectUrl
+                    };
+                    let shouldPipelineUpdate = true;
+
+                    if (
```
Comment (klu909) [inline plugins/coverage/token.js:108]: can we combine this if branch with one at the bottom?

Comment (adong) [reply plugins/coverage/token.js:108]: Can you give me one example? This is what I can come up with, only update if truly needed.

Comment (adong) [reply plugins/coverage/token.js:108]: Look in PR https://github.com/screwdriver-cd/screwdriver/pull/2947, I have made multiple attempts to optimize regards to if logic.

image

Comment (adong) [reply plugins/coverage/token.js:108]: If there's a better way, I will make another PR to optimize further

**Pull request merged**